controllers/nodehealthcheck_controller.go

```diff
@@ -881,6 +885,11 @@ func remediationFailed(remediationCR *unstructured.Unstructured, log logr.Logger
 	return succeededCondition != nil && succeededCondition.Status == metav1.ConditionFalse
 }
 
+func remediationSucceeded(remediationCR *unstructured.Unstructured, log logr.Logger) bool {
```

Comment:
Nit: I suggest converting the function into a method - that way we'll no longer need to pass it the log.

Although it's not related to this PR, I suggest the same for  `getCondition`

Reply:
I'll change also `remediationFailed` for this to work

Reply:
> that way we'll no longer need to pass it the log

this is an interesting and more complicated thing actually:
the logger on the reconciler struct is isn't request scoped, but the logger we pass around is. E.g. the latter already has the current NHC set, so it's logged every time automatically, which is very valuable. So passing the logger around makes a lot of sense.
However, I see that this isn't done everywhere, so it doesn't matter that much in the context of this PR. But it might be worth to be improved in a follow up.
Actually I wanted to explore if we can add some request scoped objects (the CR, the logger, maybe more...) to the request's context, and pass that context around everywhere... 🤔

Reply:
oh wait... you removed passing the logger around from existing functions. Please revert that. You are making logs less valuable by that.

Reply:
> the logger on the reconciler struct is isn't request scoped, but the logger we pass around is. E.g. the latter already has the current NHC set, so it's logged every time automatically, which is very valuable

I didn't consider that, thanks! 
I'll revert this change

Reply:
I might be missing something.

IIUC the value of request scoping the logger is that in essence we may use a "different" logger which might be better customized to the specific use-case we are logging.

Since *remediationSucceeded* is triggered by:
```
if r.remediationSucceeded(remediationCR, r.Log) {
...
```

It has no practical value from request scoping the logger - only the downside of making a more complicated function.

In any case I think that the correct way to use a custom logger is by creating a new business component which uses the cutomized logger instead of passing it as an argument. 

Something like [this](https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/resources/lease.go#L54):
```
func NewLeaseManager(client client.Client, holderIdent string, log logr.Logger) (LeaseManager, error) {
...
```

Reply:
**IMPORTANT**: I also just noted that I forgot to remove `remediationSucceded` from `NodeHealthcheckReconciler` functions, so I have to fix this anyway.

I think in this specific case, there is no added value, as you noted, because *I'm passing the r.Log* instead than the one created [here](https://github.com/medik8s/node-healthcheck-operator/blob/4f83bd3ad3bd794b9a3995e80aa3575693e8c26e/controllers/nodehealthcheck_controller.go#L146), and so dropping any information about the current remediation.

And `addTimeoutAnnotation` belongs already to the `NodeHealthCheckReconciler` so passing another logger as argument doesn't make sense.

For the sake of this PR, would it make sense to move the `remediationSucceded` check outside `addTimeoutAnnotation` to allow passing the "augmented" logger?

i.e.
```go
if remediationSucceded(cr, log) {
    // skip ...
} else if err := r.addTimeOutAnnotation(rm, remediationCR, now); err != nil {
    return nil, err
}
...
```

Reply:
I'm not a fan of sending logger as function/method variable.
I think that the confusion/complication it introduces overweight the value.

As I've mention before, for use cases where indeed a logger differentiation play a major role, it would usually means that a new component would be created and initialized with it's own specific logger.

In any case ATM the only additional value for this logger is the name of the remediation which can be easily added to the logging, since the remediation is already passed as a variable to `getCondition`.

Another alternative is to send the logger in the context.

To sum it up listing options by my preference.
1. Use reconciler logger without passing it as a variable
2. same as number 1, but also add to the logging the CR name (so logging will have the additional value)
3. pass a context containing the logger
4. pass the logger as variable

In any case I think this is a rather minor change so I'm not blocking anything and whatever you decide is fine by me.

Reply:
I understand your point.
IIUC, there was already [the intention to improve this logging](https://github.com/medik8s/node-healthcheck-operator/pull/318#discussion_r1562195566) and in this specific PR I'm taking the example of `remediationFailed` for the new function, so I would fix the error I noticed in the previous message and keep the logging part as is, with the intention to improving it in a follow up PR